realsense2_camera build error with librealsense2

Using [realsense](https://github.com/intel-ros/realsense/releases/tag/2.1.1) and [librealsense2](https://github.com/IntelRealSense/librealsense/releases/tag/v2.17.0)

My environment:
```
CMAKE_PREFIX_PATH=${HOME_DIR}/${WORK_DIR}/catkin_ws/devel:/opt/ros/kinetic
LD_LIBRARY_PATH=${HOME_DIR}/${WORK_DIR}/catkin_ws/devel/lib:/opt/ros/kinetic/lib:/opt/ros/kinetic/lib/x86_64-linux-gnu:/opt/Qt5.8.0/5.8/gcc_64/lib:/opt/Qt5.8.0/5.8/gcc_64/lib
PATH=/opt/ros/kinetic/bin:${HOME_DIR}/.local/bin:${HOME_DIR}/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
PKG_CONFIG_PATH=${HOME_DIR}/${WORK_DIR}/catkin_ws/devel/lib/pkgconfig:/opt/ros/kinetic/lib/pkgconfig:/opt/ros/kinetic/lib/x86_64-linux-gnu/pkgconfig
PYTHONPATH=${HOME_DIR}/${WORK_DIR}/catkin_ws/devel/lib/python3/dist-packages:/opt/ros/kinetic/lib/python2.7/dist-packages
ROS_ETC_DIR=/opt/ros/kinetic/etc/ros
ROS_PACKAGE_PATH=${HOME_DIR}/${WORK_DIR}/catkin_ws/src:/opt/ros/kinetic/share
ROS_ROOT=/opt/ros/kinetic/share/ros
```

Summary of build output
```
Scanning dependencies of target realsense2_camera
[ 84%] Building CXX object realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/realsense_node_factory.cpp.o
[ 87%] Building CXX object realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/base_realsense_node.cpp.o
[ 90%] Building CXX object realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/sr300_node.cpp.o
[ 93%] Building CXX object realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/rs415_node.cpp.o
[ 96%] Building CXX object realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/rs435_node.cpp.o
${HOME_DIR}/${WORK_DIR}/catkin_ws/src/realsense/realsense2_camera/src/base_realsense_node.cpp: In lambda function:
${HOME_DIR}/${WORK_DIR}/catkin_ws/src/realsense/realsense2_camera/src/base_realsense_node.cpp:643:56: error: ‘class rs2::processing_block’ has no member named ‘process’
                         frameset = filter_it->_filter->process(frameset);
                                                        ^
realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/build.make:86: recipe for target 'realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/base_realsense_node.cpp.o' failed
make[2]: *** [realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/src/base_realsense_node.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:1474: recipe for target 'realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/all' failed
make[1]: *** [realsense/realsense2_camera/CMakeFiles/realsense2_camera.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j8 -l8" failed
```

--- a/realsense2_camera/include/base_realsense_node.h
+++ b/realsense2_camera/include/base_realsense_node.h
@@ -54,12 +54,19 @@ namespace realsense2_camera
 
     class NamedFilter
     {
-        public:
-            std::string _name;
-            std::shared_ptr<rs2::filter> _filter;
+        private:
+#if RS2_API_VERSION >= 21700
+            typedef rs2::filter RS2_FilterType;
+#else  // RS2_API_VERSION
+            typedef rs2::processing_block RS2_FilterType;
+#endif // RS2_API_VERSION
 
         public:
-            NamedFilter(std::string name, std::shared_ptr<rs2::filter> filter):
+            std::string _name;
+            std::shared_ptr<RS2_FilterType> _filter;
+
+        public:
+            NamedFilter(std::string name, std::shared_ptr<RS2_FilterType> filter):
             _name(name), _filter(filter)
             {}
     };

--- a/realsense2_camera/include/base_realsense_node.h
+++ b/realsense2_camera/include/base_realsense_node.h
@@ -56,10 +56,10 @@ namespace realsense2_camera
     {
         public:
             std::string _name;
-            std::shared_ptr<rs2::processing_block> _filter;
+            std::shared_ptr<rs2::filter> _filter;
 
         public:
-            NamedFilter(std::string name, std::shared_ptr<rs2::processing_block> filter):
+            NamedFilter(std::string name, std::shared_ptr<rs2::filter> filter):
             _name(name), _filter(filter)
             {}
     };

